### ISSUE TITLE
WS1004

### ISSUE THREAD

**Issue by shawnbucholtz** [aws-cloudformation/cfn-lint#3103]
### CloudFormation Lint Version

cfn-lint 0.86.0

### What operating system are you using?

WSL2 Ubuntu on Windows

### Describe the bug

cfn-lint incorrectly reports:

> [cfn-lint] WS1004: Lambda function xxx does not have a corresponding log group with a Retention property

When there is a `!Ref` to a log group with a retention property.

### Expected behavior

When there is an explicit `!Ref` to a defined log group with the `RetentionInDays` property set then WS1004 should not be flagged.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform: AWS::Serverless-2016-10-31

Resources:

  LambdaProxyFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: src/
      FunctionName: MyFunction
      Handler: index.handler
      LoggingConfig:
        LogFormat: JSON
        LogGroup: !Ref LambdaProxyLogGroup
      MemorySize: 512
      PackageType: Zip
      ReservedConcurrentExecutions: 1
      Runtime: nodejs18.x
      Timeout: 10
      Tracing: Active

  LambdaProxyLogGroup:
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain
    Type: AWS::Logs::LogGroup
    Properties:
      LogGroupClass: STANDARD
      RetentionInDays: 180
```

**Comment by PatMyron** [comment]
`cfn-lint` itself does not vend `WS1004`. Seems to be from this popular rule pack: 
https://awslabs.github.io/serverless-rules/rules/
https://github.com/awslabs/serverless-rules/issues
